### PR TITLE
Code cleanup and refactoring

### PR DESCRIPTION
**Pull request by ywwg**
* Add support for handling SIGTERM signal
* Refactor Run function into api module instead of binary so it can be called from e2e tests without needing to start up the binary itself.
* Add wrappers for error and warning logging so we don't need to keep ignoring the return value

### CODE REVIEW

**Comment by carrieedwards** [comment]
I think the sigterm handling needs to change a bit if we use the pkg/server instead of the weaveworks/common server, since pkg/server doesn't have its own SignalHandler interface

**Comment by ywwg** [comment]
ok I'll update after your change goes in

**Comment by ywwg** [comment]
hand-tested on the command line, works

**Review by leizor — DISMISSED** [review]
I like the new logging layer!

File: pkg/util/log/log.go [hunk 1]
```diff
@@ -61,3 +61,15 @@ func CheckFatal(location string, err error) {
 		os.Exit(1)
 	}
 }
+
+// Error logs an error and panics if the log call itself had an error.
+func Error(logger log.Logger, keyvals ...interface{}) {
+	if err := level.Error(logger).Log(keyvals); err != nil {
+		panic(fmt.Sprintf("error writing to log: %v", err))
```
[the remark this answers is not in the document]
Comment (ywwg) [reply pkg/util/log/log.go:68]: oh yeah I can do that

**Comment by ywwg** [comment]
oh hm %w doesn't work in sprintf

File: pkg/util/log/log.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ywwg) [reply pkg/util/log/log.go:68]: actually I can't! %w is only supported in fmt.Errorf

[the remark this answers is not in the document]
Comment (leizor) [reply pkg/util/log/log.go:68]: Ohh gotcha, interesting

**Comment by ywwg** [comment]
looks like I broke tests so I'll do that after lunch

**Review by carrieedwards — APPROVED** [review]
Looks great!

**Pull request merged**